holidays/countries/ethiopia.py

```diff
@@ -100,14 +83,19 @@ def _populate_public_holidays(self):
             self._add_holiday_may_28(tr("ደርግ የወደቀበት ቀን"))
 
         # Ethiopian New Year.
-        self._add_holiday(tr("የዘመን መለወጫ (እንቁጣጣሽ) በዓል"), SEP, 12 if self._is_leap_year() else 11)
+        self._add_ethiopian_new_year(tr("የዘመን መለወጫ (እንቁጣጣሽ) በዓል"))
 
         # Finding of True Cross.
-        self._add_holiday(tr("የመስቀል በዓል"), SEP, 28 if self._is_leap_year() else 27)
+        self._add_finding_of_true_cross(tr("የመስቀል በዓል"))
 
         if self._year <= 1990:
-            # Popular Revolution Commemoration Day.
-            self._add_holiday(tr("የአብዮት ቀን"), SEP, 13 if self._is_leap_year() else 12)
+            # Julian Date Drift shouldn't affect this one.
+            self._add_holiday(
+                # Popular Revolution Commemoration Day.
+                tr("የአብዮት ቀን"),
+                SEP,
+                13 if is_ethiopian_leap_year(self._year) else 12,
+            )
```

Comment:
If this holiday continued to exist, it could be implemented as [ethiopian_new_year + 1 day].